pypeit/spectrographs/spectrograph.py

```diff
@@ -110,19 +105,32 @@ def __init__(self):
     def default_pypeit_par():
         return pypeitpar.PypeItPar()
 
-    def nonlinear_counts(self, det=1):
+    def nonlinear_counts(self, det, datasec_img=None):
```

Comment:
Thanks for implementing this. See my suggestion above for what the processrawimage object should include. I think non-linear counts should be an attribute for the object that holds the images.